### DISCUSSION TITLE
How to convert "../NewPage" route to nalu navigation?

### DISCUSSION THREAD

**Discussion by CodingOctocat** [nalu-development/nalu#54] [Q&A] [answered]
I don't quite understand how to navigate, my navigation path is: **`Root` -> `CreateNewPage` -> `NewPage`**, and when the `NewPage` is backed up, it should go back to the `Root` (i.e.: `CreateNewPage` should not keep itself in the stack when it navigates to the `NewPage`).

CreateNewPage.xaml.cs

```cs
await Shell.Current.GoToAsync($"../{Routes.NewPage}", navigationParameter);

// Is this correct?
await _navigationService.GoToAsync(Navigation.Relative().Pop().Push<NewPageViewModel>().WithIntent(navigationParameter));
```

**Comment by albyrock87** [comment, marked as answer]
@CodingOctocat if 
- `RootPage` is your `ShellContent`
-  you're currently in `RootPage` -> `CreateNewPage` -> `NewPage` 
- your goal is to go into `RootPage` -> `SomeOtherPage`

then what you can do is either:
- `Navigation.Relative().Pop().Pop().Push<SomeOtherViewModel>()`
- `Navigation.Absolute().ShellContent<RootViewModel>().Push<SomeOtherViewModel>()`
